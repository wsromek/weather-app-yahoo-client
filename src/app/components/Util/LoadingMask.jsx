import React from 'react';
import PropTypes from 'prop-types';

function LoadingMask({ loading = null }) {
  if (!loading) return null;

  return (
    <div className="loading-overlay">
      <img alt="loading_indicator" src="/public/images/throbber.gif" />
    </div>
  );
}

LoadingMask.propTypes = {
  loading: PropTypes.bool.isRequired,
};

export default LoadingMask;
