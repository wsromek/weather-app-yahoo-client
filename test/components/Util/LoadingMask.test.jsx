import React from 'react';
import renderer from 'react-test-renderer';
import LoadingMask from '../../../src/app/components/Util/LoadingMask';

describe('LoadingMask component', () => {
  it('renders correctly', () => {
    const present = renderer.create(
      <LoadingMask loading />,
    ).toJSON();

    const absent = renderer.create(
      <LoadingMask loading={false} />,
    ).toJSON();

    expect(present).toMatchSnapshot();
    expect(absent).toMatchSnapshot();
  });
});
