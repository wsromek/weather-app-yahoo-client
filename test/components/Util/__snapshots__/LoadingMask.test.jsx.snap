// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMask component renders correctly 1`] = `
<div
  className="loading-overlay"
>
  <img
    alt="loading_indicator"
    src="/public/images/throbber.gif"
  />
</div>
`;

exports[`LoadingMask component renders correctly 2`] = `null`;
